compiler-rt/lib/radsan/tests/radsan_test_interceptors.cpp

```diff
@@ -236,6 +240,72 @@ TEST(TestRadsanInterceptors, readDiesWhenRealtime) {
   std::remove(temporary_file_path());
 }
 
+TEST(TestRadsanInterceptors, writeDiesWhenRealtime) {
+  auto fd = open(temporary_file_path(), O_WRONLY);
+  EXPECT_THAT(fd, Ne(-1));
+  auto func = [fd]() {
+    char c{};
+    write(fd, &c, 1);
+  };
+  expectRealtimeDeath(func, "write");
+  expectNonrealtimeSurvival(func);
+  close(fd);
+  std::remove(temporary_file_path());
+}
+
+#if SANITIZER_APPLE
+TEST(TestRadsanInterceptors, preadDiesWhenRealtime) {
+  auto fd = open(temporary_file_path(), O_RDONLY);
+  auto func = [fd]() {
```

Comment:
Presumably these files still have the issue as well, these tests don't fail because mac seems more resilient to writing to invalid fds. I think in an ideal world that validity check would exist in these other calls as well